internal/controller/appconfigurationprovider_controller.go

```diff
@@ -198,18 +199,28 @@ func (reconciler *AzureAppConfigurationProviderReconciler) Reconcile(ctx context
 		}
 	}
 
+	if reconciler.ProvidersReconcileState[req.NamespacedName].ClientManager == nil ||
+		reconciler.ProvidersReconcileState[req.NamespacedName].Generation != provider.Generation {
+		manager, err := loader.NewConfigurationClientManager(ctx, *provider)
+		if err != nil {
+			reconciler.logAndSetFailStatus(ctx, err, provider)
+			return reconcile.Result{Requeue: true, RequeueAfter: RequeueReconcileAfter}, nil
+		}
+		reconciler.ProvidersReconcileState[req.NamespacedName].ClientManager = manager
+	}
+
 	/* Create ConfigurationSettingLoader to get the key-value settings from Azure AppConfiguration. */
-	configProvider, err := loader.NewConfigurationSettingLoader(ctx, *provider, nil)
+	manager := reconciler.ProvidersReconcileState[req.NamespacedName].ClientManager
```

Comment:
Put this line before line 202

In line 202, just judge whether `manager == nil`